File: app/src/main/java/blue/koenig/kingsfamily/dagger/OverviewAppComponent.java
package blue.koenig.kingsfamily.dagger;

import javax.inject.Singleton;

import blue.koenig.kingsfamily.view.OverviewActivity;
import blue.koenig.kingsfamilylibrary.dagger.AppModule;
import blue.koenig.kingsfamilylibrary.dagger.ConnectionModule;
import blue.koenig.kingsfamilylibrary.dagger.FamilyAppComponent;
import blue.koenig.kingsfamilylibrary.dagger.InstallationModule;
import dagger.Component;

/**
 * Created by Thomas on 17.09.2017.
 */
@Singleton
@Component(modules = {AppModule.class, ConnectionModule.class, InstallationModule.class})
public interface OverviewAppComponent extends FamilyAppComponent {
    void inject(OverviewActivity target);
}

File: app/src/main/java/blue/koenig/kingsfamily/dagger/OverviewAppComponent.kt
package blue.koenig.kingsfamily.dagger

import blue.koenig.kingsfamily.view.OverviewActivity
import blue.koenig.kingsfamilylibrary.dagger.*
import dagger.Component
import javax.inject.Singleton

/**
 * Created by Thomas on 17.09.2017.
 */
@Singleton
@Component(modules = [(AppModule::class), (ConnectionModule::class), (InstallationModule::class), (ModelModule::class)])
interface OverviewAppComponent : FamilyAppComponent {
    fun inject(target: OverviewActivity)
}
